apps/src/pythonlab/pythonHelpers/messageHelpers.ts

```diff
@@ -41,10 +56,56 @@ export function parseErrorMessage(errorMessage: string) {
   }
   if (mainErrorLine >= errorLines.length) {
     // If we never find the main.py error, return the entire message.
-    return errorMessage;
+    return neighborhoodExceptionMessage
+      ? `${neighborhoodExceptionMessage}\n${errorMessage}`
+      : errorMessage;
+  }
+  const adjustedErrorLines = errorLines
+    .slice(mainErrorLine, errorLines.length)
+    .join('\n');
+
+  return neighborhoodExceptionMessage
+    ? `${neighborhoodExceptionMessage}\n${adjustedErrorLines}`
+    : adjustedErrorLines;
+}
+
+/**
+ * This method returns the user-friendly message mapped to a Neighborhood exception type key.
+ * @param exceptionType
+ */
+export function getNeighborhoodExceptionMessage(exceptionType: string) {
+  if (exceptionType in NeighborhoodExceptionType) {
+    // Return the user-friendly error message mapped to the NeighborhoodExceptionType.
+    const message =
+      NeighborhoodExceptionMessage[exceptionType as NeighborhoodExceptionType];
+    return `${commonI18n.exceptionTag()} ${message}`;
+  } else {
+    return `[EXCEPTION] Unknown Neighborhood exception: ${exceptionType}`;
```

Comment:
nit: use translation for `[EXCEPTION]` (and also add one for "Unknown Neighborhood exception"?)

Reply:
Updated!